src/Raven.Studio/typescript/viewmodels/resources/widgets/ongoingTasksWidget.ts

```diff
@@ -0,0 +1,202 @@
+            case "SubscriptionCount": return "Subscription";
+        }
+    }
+}
+
+class nodeRawData {
+    nodeTag = ko.observable<string>();
+    databaseItems = ko.observableArray<Raven.Server.Dashboard.DatabaseOngoingTasksInfoItem>([]);
+}
+
+class ongoingTasksWidget extends websocketBasedWidget<Raven.Server.Dashboard.Cluster.Notifications.OngoingTasksPayload> {
+    
+    allNodesData = ko.observableArray<nodeRawData>();
+    
+    taskList = ko.observableArray<ongoingTaskItem>([]);
+
+    getType(): Raven.Server.Dashboard.Cluster.ClusterDashboardNotificationType {
+        return "OngoingTasks";
+    }
+
+    constructor(controller: clusterDashboard) {
+        super(controller);
+        
+        this.taskList([
+            new ongoingTaskItem("ExternalReplicationCount", 0, "external-replication", "icon-external-replication"),
+            new ongoingTaskItem("ReplicationHubCount", 0, "replication-hub", "icon-pull-replication-hub"),
+            new ongoingTaskItem("ReplicationSinkCount", 0, "replication-sink", "icon-pull-replication-agent"),
+            new ongoingTaskItem("RavenEtlCount", 0, "ravendb-etl", "icon-ravendb-etl"),
+            new ongoingTaskItem("OlapEtlCount", 0, "olap-etl", "icon-olap-etl"),
+            new ongoingTaskItem("SqlEtlCount", 0, "sql-etl", "icon-sql-etl"),
+            new ongoingTaskItem("PeriodicBackupCount", 0, "periodic-backup", "icon-backups"),
+            new ongoingTaskItem("SubscriptionCount", 0, "subscription", "icon-subscription")
+        ]);
+    }
+
+    compositionComplete() {
+        super.compositionComplete();
+        this.enableSyncUpdates();
+
+        for (let ws of this.controller.getConnectedLiveClients()) {
+            this.onClientConnected(ws);
+        }
+    }
+
+    onData(nodeTag: string, data: Raven.Server.Dashboard.Cluster.Notifications.OngoingTasksPayload) {
+        
+        const nodeToUpdate = this.allNodesData().find(node => node.nodeTag() === nodeTag);
+
+        if (nodeToUpdate) {
+            data.Items.forEach(dbInfo => {
+                const filteredArray = nodeToUpdate.databaseItems().filter(x => x.Database !== dbInfo.Database);
+                filteredArray.push(dbInfo);
+                nodeToUpdate.databaseItems(filteredArray);
+            });
+        } else {
+            const nodeToUpdate = new nodeRawData();
+            nodeToUpdate.nodeTag(nodeTag);
+            data.Items.forEach(dbInfo => nodeToUpdate.databaseItems().push(dbInfo));
+            this.allNodesData().push(nodeToUpdate);
+        }
+        
+        // at this point allNodesData is updated - now update UI classes
+        this.initTaskList();
+        
+        this.allNodesData().forEach(node => {
+            for (let i = 0; i < this.taskList().length; i++) {
+                this.sumTaskCount(node, this.taskList()[i])
+            }
+        });
+    }
+    
+    private sumTaskCount(node: nodeRawData, taskItem: ongoingTaskItem): void {
+        let mustAddNode = false;
```

Comment:
I think we don't need it - just use task count from task item.

Reply:
Cannot really do that - let me try to explain  
`sumTaskCount` is called in a loop on all nodes
and then we loop on the per-node inner array items from the raw data

If I add the node tag only according to the `taskItem.taskCount` then you get wrong nodes, i.e. :

If raw data has:

```
Node A 
    db1 
         [extRep 1, Backp 2, ....]
    db2
         [ ... ]

Node B
    db1
         [extRep 0, Backup 0, ...]
    db2
         [ ... ]  
```            
         
then both node A & node B will be shown in the table
because we look at task.count with is the aggregation
but we need to show only A         

so I only want to really add a node if `countToAdd is > 0`